File: from-2801-to-2900/2900-longest-unequal-adjacent-groups-subsequence-i/solution.go
func getLongestSubsequence(words []string, groups []int) []string {
    n := len(groups)

    getAlternative := func(startVal int) []int {
        idxs := []int{}
        i := 0
        for i < n {
            if groups[i] == startVal {
                idxs = append(idxs, i)
                break
            }
            i += 1
        }
        lastVal := startVal
        for i < n {
            if groups[i] != lastVal {
                idxs = append(idxs, i)
                lastVal = groups[i]
            }
            i += 1
        }
        return idxs
    }

    fromOne := getAlternative(1)
    fromZero := getAlternative(0)
    
    longest := fromOne
    if len(fromZero) > len(fromOne) {
        longest = fromZero
    }

    result := []string{}
    for _, idx := range longest {
        result = append(result, words[idx])
    }
    return result
}

File: from-2801-to-2900/2900-longest-unequal-adjacent-groups-subsequence-i/solution.py
class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        idxs = [0]
        for i in range(1, len(groups)):
            if groups[i] != groups[i - 1]:
                idxs.append(i)

        return [words[i] for i in idxs]